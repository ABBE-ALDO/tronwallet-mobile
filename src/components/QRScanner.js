import React from 'react'
import QRCodeScanner from 'react-native-qrcode-scanner'
import Ionicons from 'react-native-vector-icons/Ionicons'
import PropTypes from 'prop-types';
import { Colors } from '../components/DesignSystem'
import * as Utils from '../components/Utils'

const QRCodeComponent = ({ onRead, onClose }) => (
  <Utils.Container>
    <Utils.Header>
      <Utils.TitleWrapper>
        <Utils.Title>Address Scanner</Utils.Title>
      </Utils.TitleWrapper>
      <Utils.CloseButton onPress={onClose}>
        <Ionicons name='ios-close-circle-outline' size={32} color={Colors.primaryText} />
      </Utils.CloseButton>
    </Utils.Header>
    <QRCodeScanner
      showMarker
      reactivate
      fadeIn
      customMarker={(
        <Utils.View flex={1} background='transparent' justify='center' align='center'>
          <Utils.View width={250} height={250} borderWidth={2} borderColor={'white'} />
          <Utils.Text marginTop='medium' align='center'>Scan the QRCode to identify the target user</Utils.Text>
        </Utils.View>
      )}
      cameraStyle={{ height: '100%', width: '100%', justifyContent: 'flex-start' }}
      permissionDialogMessage="To scan the public key the app needs your permission to access the camera."
      onRead={onRead}
    />
  </Utils.Container>
)

QRCodeComponent.propTypes = {
  onRead: PropTypes.func.isRequired
}

export default QRCodeComponent