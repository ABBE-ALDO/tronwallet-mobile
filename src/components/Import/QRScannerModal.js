import React from 'react'
import { Modal } from 'react-native'
import { bool, func } from 'prop-types'

import QRScanner from '../QRScanner'

const QRScannerModal = ({ visible, onClose, onRead }) => (
  <Modal
    visible={visible}
    onRequestClose={onClose}
    animationType='slide'
  >
    <QRScanner
      onRead={onRead}
      onClose={onClose}
      checkAndroid6Permissions
    />
  </Modal>
)

QRScannerModal.propTypes = {
  visible: bool,
  onClose: func.isRequired,
  onRead: func.isRequired
}

QRScannerModal.defaultProps = {
  visible: false
}

export default QRScannerModal
