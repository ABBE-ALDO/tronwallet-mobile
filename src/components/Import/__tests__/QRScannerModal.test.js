import React from 'react'
import { shallow } from 'enzyme'
import { Modal } from 'react-native'

import QRScanner from '../../QRScanner'
import QRScannerModal from '../QRScannerModal'

jest.mock('react-native-qrcode-scanner')
jest.mock('../../../utils/i18n')

describe('QR Scanner Modal component', () => {
  const setup = propOverrides => {
    const props = Object.assign({
      onClose: () => {},
      onRead: () => {}
    }, propOverrides)

    const wrapper = shallow(
      <QRScannerModal {...props} />
    )

    const modalWrapper = wrapper.find(Modal)

    return {
      props,
      wrapper,
      modalWrapper,
      qrScannerWrapper: modalWrapper.find(QRScanner)
    }
  }

  test('Should match when render component', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  test('Should not show modal when visible is false by default', () => {
    const { modalWrapper } = setup()
    expect(modalWrapper.prop('visible')).toBe(false)
  })

  test('Should show modal when visible is true', () => {
    const { modalWrapper } = setup({ visible: true })
    expect(modalWrapper.prop('visible')).toBe(true)
  })

  test('Should onClose triggered when close modal', () => {
    const onClose = jest.fn()
    const { modalWrapper } = setup({ onClose })

    const { onRequestClose } = modalWrapper.props()
    onRequestClose()

    expect(onClose).toBeCalled()
  })

  test('Should onClose triggered when close QRScanner', () => {
    const onClose = jest.fn()
    const { qrScannerWrapper } = setup({ onClose })

    const closeQRScanner = qrScannerWrapper.prop('onClose')
    closeQRScanner()

    expect(onClose).toBeCalled()
  })

  test('Should onRead triggered when QR was scanned', () => {
    const onRead = jest.fn()
    const { qrScannerWrapper } = setup({ onRead })

    const readQRScanner = qrScannerWrapper.prop('onRead')
    readQRScanner()

    expect(onRead).toBeCalled()
  })
})
