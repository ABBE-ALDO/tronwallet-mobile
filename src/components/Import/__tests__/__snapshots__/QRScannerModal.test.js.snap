// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QR Scanner Modal component Should match when render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QRScannerModal
    onClose={[Function]}
    onRead={[Function]}
    visible={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animationType": "slide",
      "children": <QRCodeComponent
        checkAndroid6Permissions={true}
        onClose={[Function]}
        onRead={[Function]}
      />,
      "hardwareAccelerated": false,
      "onRequestClose": [Function],
      "visible": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "checkAndroid6Permissions": true,
        "onClose": [Function],
        "onRead": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animationType": "slide",
        "children": <QRCodeComponent
          checkAndroid6Permissions={true}
          onClose={[Function]}
          onRead={[Function]}
        />,
        "hardwareAccelerated": false,
        "onRequestClose": [Function],
        "visible": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checkAndroid6Permissions": true,
          "onClose": [Function],
          "onRead": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
